crypto/bio/bio_sock.c

```diff
@@ -256,10 +256,10 @@ int BIO_get_accept_socket(char *host, int bind_mode)
     BIO_ADDRINFO *res = NULL;
 
     if (!BIO_parse_hostserv(host, &h, &p, BIO_PARSE_PRIO_SERV))
-        return INVALID_SOCKET;
+        goto err;
```

Comment:
Note that similar changes would be needed also for several other uses of `BIO_parse_hostserv()`.
Rather than trying to do this change everywhere (which is not possible for uses outside of OpenSSL),
`BIO_parse_hostserv()` should be fixed such that on error it deallocates its output values and returns NULL in the respective output parameters.